src/request/sign-swap/SignSwapApi.js

```diff
@@ -218,6 +257,80 @@ class SignSwapApi extends BitcoinRequestParserMixin(TopLevelApi) {
         return /** @type KeyguardRequest.SignSwapRequestLayout */ (layout);
     }
 
+    /**
+     *
+     * @param {unknown} direction
+     * @returns {'left-to-right' | 'right-to-left'}
+     */
+    parseDirection(direction) {
+        if (typeof direction !== 'string' || (direction !== 'left-to-right' && direction !== 'right-to-left')) {
+            throw new Error('Invalid direction');
+        }
+        return direction;
+    }
+
+    /**
+     *
+     * @param {KeyguardRequest.PolygonTransactionInfo} request
+     * @param {string[]} allowedMethods
```

Comment:
```suggestion
     * @param {Array<'open' | 'openWithApproval' | 'redeem' | 'redeemWithSecretInData'>} allowedMethods
```
Only a subset of possible methods is supported by this method.